xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-test/xwiki-platform-flamingo-skin-test-docker/src/test/it/org/xwiki/flamingo/test/docker/RenamePageIT.java

```diff
@@ -668,4 +665,56 @@ void renameWithIndexingWaiting(String strategy, TestUtils setup, TestReference t
             setup.rest().delete(otherReference);
         }
     }
+
+    @Order(9)
+    @Test
+    void renameWithRelativeLinks(TestUtils testUtils, TestReference testReference, TestConfiguration testConfiguration)
+        throws Exception
+    {
+        testUtils.createPage(testReference, "[[Alice]]\n[[Bob]]\n[[Eve]]", "Test relative links");
+        SpaceReference rootSpaceReference = testReference.getLastSpaceReference();
+        SpaceReference aliceSpace = new SpaceReference("Alice", rootSpaceReference);
+        testUtils.createPage(new DocumentReference("WebHome", aliceSpace), "Alice page",
+            "Alice");
+        SpaceReference bobSpace = new SpaceReference("Bob", rootSpaceReference);
+        testUtils.createPage(new DocumentReference("WebHome", bobSpace), "[[Alice]]",
+            "Alice");
+
+        // Wait for an empty queue here to ensure that the deleted page has been removed from the index and links
+        // won't be updated just because the page is still in the index.
+        new SolrTestUtils(testUtils, testConfiguration.getServletEngine()).waitEmptyQueue();
```

Comment:
This should not be needed anymore with XWIKI-22323.

Reply:
I actually call it to avoid getting a question when performing the rename operation.

Reply:
The question is supposed to automatically vanish as soon as the job is done AFAIU.

Reply:
well what's sure is that I'm getting a timeout when I remove this and the screenshot shows the question...

Reply:
I think the problem is more that indexing takes quite some time, in particular with the frequent committing of the index with the default parameters, and the page object most likely just doesn't wait long enough. I have an integration test that explicitly triggers the question and there it works to just continue waiting without answering the question.